kumascript/macros/CSSRef.ejs

```diff
@@ -814,42 +814,116 @@ function smartLink(href, title, content, subpath) {
+}
+
 function buildSublist(pages, title) {
   let result = `<li class="toggle"><details><summary>${title}</summary><ol>`;
 
   for (const page of pages) {
-    const url = page.url.replace('en-US', locale);
-    const title = htmlEscape(page.title);
-
-    if (locale != 'en-US') {
-      page.translations.forEach(function(translation){
-        if (translation.locale === locale) {
-          title = htmlEscape(translation.title);
-        }
-      });
-    }
+     result += makePageLink(page);
+  }
 
-    result += '<li>';
+  result += '</ol></details></li>';
 
-    if (hasTag(page, 'Experimental')) {
-      result += badges.ExperimentalBadge;
-    }
+  return result;
+}
 
-    if (hasTag(page, 'Deprecated')) {
-      result += badges.DeprecatedBadge;
-    }
+/**
+ * Special case behavior for CSS properties. Instead of just listing them all,
+ * put properties with common prefixes (like `border-`) in collapsible groups.
+ *
+ * @param propertyGroups : an object returned by a call to groupProperties
+ * @param pages : an array of page objects
+ * @param title : the title of this section of the sidebar
+ * @returns : markup for the complete CSS property section
+ */
+function buildPropertylist(pages, title) {
 
-    if (rtlLocales.indexOf(locale) != -1) {
-      result += '<bdi>';
-    }
+  let result = `<li class="toggle"><details><summary>${title}</summary><ol>`;
 
-    result += smartLink(url, null, title, cssURL);
+  // Go through the array of property pages, splitting off the "first part"
+  // of each property name, where "parts" are demarcated by "-".
+  // For example, split "border" off "border-style".
+  //
+  // Make an object:
+  // - with a key for every distinct "first part"
+  // - whose value is an array of all CSS property names which start with that part.
+  //
+  // For example, it has a key "border" whose value is an array of all
+  // the CSS properties that start with "border".
+  const propertyGroups = {};
 
-    if (rtlLocales.indexOf(locale) != -1) {
-      result += '</bdi>';
+  for (const property of properties) {
+    // The regex is: match at "-" but not at the start of the string,
+    // because of properties like -webkit-line-clamp, where we want the
+    // "first part" to be "-webkit"
+    const pieces = property.title.split(/(?<!^)-/);
+    if (Object.keys(propertyGroups).includes(pieces[0])) {
+      propertyGroups[pieces[0]].push(property.title);
+    } else {
+      propertyGroups[pieces[0]] = [property.title];
     }
+  }
 
-    result += '</li>';
+  for (const propertyGroupName of Object.keys(propertyGroups)) {
+    const group = propertyGroups[propertyGroupName];
+    if (group.length < 3) {
+      // if a group contains fewer than 3 entries (that is, there are fewer than 3
+      // properties with the given prefix) then just write them out as a list
+      // of links
+      for (const propertyName of group) {
+        result += makePageLink(pages.find(item => item.title === propertyName));
+      }
```

Comment:
If we put the pages directly into the groups above, we don't need to find them here (i.e. better performance):

```suggestion
      for (const page of group) {
        result += makePageLink(page);
      }
```